### PR TITLE
Buffer loading of stars.dat and cross index files

### PR DESCRIPTION
**Pull request by ajtribick**
In my testing this roughly halved the time taken to load the stars.dat file.

### CODE REVIEW

File: src/celengine/stardb.cpp [hunk 1]
```diff
@@ -60,8 +61,8 @@ constexpr inline float STAR_OCTREE_ROOT_SIZE   = 1000000000.0f;
 constexpr inline float STAR_OCTREE_MAGNITUDE   = 6.0f;
 //constexpr const float STAR_EXTRA_ROOM        = 0.01f; // Reserve 1% capacity for extra stars
 
-constexpr inline std::string_view FILE_HEADER            = "CELSTARS"sv;
-constexpr inline std::string_view CROSSINDEX_FILE_HEADER = "CELINDEX"sv;
+constexpr inline std::string_view FILE_IDENTIFIER            = "CELSTARS"sv;
```
Comment (375gnu) [inline src/celengine/stardb.cpp:64]: it seems that using `inline` makes sense for header files only and it's noop for cpp files.

File: src/celengine/stardb.cpp [hunk 2]
```diff
@@ -77,6 +78,15 @@ constexpr inline AstroCatalog::IndexNumber TYC3_MAX   = 3u;     // from TYC2
 constexpr inline AstroCatalog::IndexNumber TDSC_TYC3_MAX            = 4u;
 constexpr inline AstroCatalog::IndexNumber TDSC_TYC3_MAX_RANGE_TYC1 = 2907u;
 
+// stars.dat file header and record sizes
+constexpr inline std::size_t FILE_HEADER_SIZE = FILE_IDENTIFIER.size() + sizeof(std::uint16_t) + sizeof(std::uint32_t);
+constexpr inline std::size_t FILE_RECORD_SIZE = sizeof(AstroCatalog::IndexNumber)
+                                              + 3 * sizeof(float)
+                                              + sizeof(std::int16_t) + sizeof(std::uint16_t);
```
Comment (375gnu) [inline src/celengine/stardb.cpp:85]: This looks too fragile. What if we introduce structures for this purpose?
```
struct __attribute__((__packed__)) StarDBHeader
{
    unsigned char id[8];
    std::uint16_t version;
    std::uint32_t counter;
};

// and

struct __attribute__((__packed__)) StarDBRecord
{
    AstroCatalog::IndexNumber indexNumber;
    float x, y, z;
    std::int16_t absMag;
    std::uint16_t spectralType;
};
```

**Review by 375gnu — CHANGES_REQUESTED** [review]

File: src/celengine/stardb.cpp (hunk 1 above)
Comment (ajtribick) [reply src/celengine/stardb.cpp:64]: Yep, I tend to prefer to keep it though in case of needing to move this to a header file later.

**Comment by 375gnu** [comment]
And also another question: doesn't make sense to use a bigger buffer (e.g. `lcd(4096, sizeof(StarDBRecord))`) instead of reading records one by one. I think this can boost loading further.

File: src/celengine/stardb.cpp (hunk 2 above)
Comment (ajtribick) [reply src/celengine/stardb.cpp]: Done

**Review by 375gnu — APPROVED** [review]

**Pull request merged**